test/unit-test/ota_utest.c

```diff
@@ -902,7 +959,26 @@ void test_OTA_ReceiveFileBlockEmpty()
     TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
 }
 
-void test_OTA_ReceiveFileBlockComplete()
+void test_OTA_ReceiveFileBlockTooLarge()
+{
+    OtaEventMsg_t otaEvent = { 0 };
+
+    pOtaJobDoc = JOB_DOC_HTTP;
+
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForFileBlock, OTA_GetState() );
+
+    otaInterfaces.os.event.send = mockOSEventSend;
+
+    otaEvent.eventId = OtaAgentEventReceivedFileBlock;
+    otaEvent.pEventData = &eventBuffer;
+    otaEvent.pEventData->dataLength = 2 * OTA_FILE_BLOCK_SIZE;
```

Comment:
It may be better to set this to the block size + 1. Setting it to 2x the expected block size doesn't check for off by one bugs in the code.